deploy/crds/nodemaintenance_crd.yaml

```diff
@@ -17,23 +16,51 @@ spec:
     openAPIV3Schema:
       properties:
         apiVersion:
+          description: 'APIVersion defines the versioned schema of this representation
+            of an object. Servers should convert recognized schemas to the latest
+            internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#resources'
           type: string
         kind:
+          description: 'Kind is a string value representing the REST resource this
+            object represents. Servers may infer this from the endpoint the client
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds'
           type: string
         metadata:
           type: object
         spec:
           properties:
             nodeName:
+              description: Node name to apply maintanance on/off
               type: string
             reason:
+              description: Reason for maintanance
               type: string
           required:
           - nodeName
           type: object
         status:
           properties:
-            phase:
+            completed:
```

Comment:
Was a Status.Phase enum considered to indicate things like Running|Succeeded|Failed?

That would be more consistent with how many of our other apis (and the k8s api) is structured.

Reply:
I can agree to that.  
I'll remove the completed boolean field in the next commit and return phase with these values.
one thing to keep in mind is that this field can move from Failed to Running/Succeeded in sequential reconciliation loops.